src/hip_hcc.cpp

```diff
@@ -2514,6 +2515,14 @@ hipError_t hipHccGetAcceleratorView(hipStream_t stream, hc::accelerator_view** a
 // TODO - add a contect sequence number for debug. Print operator<< ctx:0.1 (device.ctx)
 
 namespace hip_impl {
+    std::unordered_set<std::string> get_all_gpuarch() {
+        std::unordered_set<std::string> r{};
```

Comment:
For inserting and searching on a small set of data, set should give you a slight better performance against unordered_set

http://quick-bench.com/LZsAxAb4LGq5AjOSZeiQFun4oz8

Reply:
Everytime `get_all_gpuarch` is called, it constructs a new `set`.  One option would be to have a static `set` in `get_all_gpuarch`, which would get initialized once and make the function to return its reference instead.

Reply:
done.

Reply:
> For inserting and searching on a small set of data, set should give you a slight better performance against unordered_set
> 
> http://quick-bench.com/LZsAxAb4LGq5AjOSZeiQFun4oz8

That suggested benchmark is not showing what you think it does. Associative data structures are interesting because they allow more efficient searching than a linear search on their underlying linked data structure. Furthermore, there are no conflicts, and conflict resolution is an important aspect here. Finally, a vector is almost always going to be faster.